mtags/src/main/scala-3/scala/meta/internal/pc/InferredTypeProvider.scala

```diff
@@ -251,4 +255,32 @@ final class InferredTypeProvider(
     end match
   end inferredTypeEdits
 
+  private def namePos(text: String, tree: untpd.NameTree)(using
+      Context
+  ): SourcePosition =
+    val realName = tree.name.stripModuleClassSuffix.lastPart.toString.toList
+
+    def lookup(
+        idx: Int,
+        start: Option[(Int, List[Char])],
```

Comment:
[note]
A pair of next position and trailing realName

Reply:
Maybe forge this note into code?
```scala
case class Lookup(position: Int, tralingName: List[Char])
```